infra/asttools.py

```diff
@@ -76,11 +76,16 @@ def iter_methods(cls):
 inline_cache = {}
 def load(filename):
     if filename not in inline_cache:
+        inline_cache[filename] = "in-progress"
         with open(filename) as f:
             ast = parse(f.read(), filename)
         ast = inline(ast)
         inline_cache[filename] = ast
-    return inline_cache[filename]
+        return ast
+    elif inline_cache[filename] == "in-progress":
```

Comment:
Is it also easy to check for unused imports?

Reply:
Huh. I don’t think so, since this is only looking at imports, not uses.